### DISCUSSION TITLE
how to deal with byte array mapping ?

### DISCUSSION THREAD

**Discussion by beeondev** [openapi-processor/openapi-processor-spring#191] [Q&A] [answered]
suppose i have a openapi.yaml whith  : 

```
    ...
    responses:
        "200":
          description: OK
          content:
            application/octet-stream:
              schema:
                 type: string
                 format: binary 
      ...
``` 
What i want is to build mapping.yaml in order to return a Mono<ResponseEntity<byte[]>>.

How can i achieve that ?
(this is because i have as legacy code signature i do not want to break)

**Comment by hauner** [comment, marked as answer]
looks like `Mono<ResponseEntity<byte[]>>` isn't something the processor can generate at the moment. The mapping parser doesn't handle primitive types. 

a workaround is to exclude the endpoint

    map:
      paths:
        /my-endpoint:
          exclude: true

and write the controller method yourself.

or 

you could try to replace the `byte[]` with something like `java.nio.ByteBuffer` (as an endpoint specific mapping). But I don't know if Spring can serialize this out of the box.   

    map:
      paths:
        /my-endpoint:
          types:
           - type: string:binary => java.nio.ByteBuffer

to generate the `Mono` and `ResponseEntity` you need:

    map:
      result: org.springframework.http.ResponseEntity

      single: reactor.core.publisher.Mono
      multi: reactor.core.publisher.Flux

note that the processor generates `ResponseEntity<Mono<..>>`. It should still work.

If  for some reason it must be `Mono<ResponseEntity<..>>` you could switch the `result` and `single` mapping (on the endpoint level), but that is a bit hacky..    

    paths:
      /my-endoint:
        result: reactor.core.publisher.Mono
        single: org.springframework.http.ResponseEntity

**Comment by hauner** [comment]
openapi-processor/openapi-processor-base#59
